webhook_features/src/main.rs

```diff
@@ -71,11 +73,17 @@ fn features() -> Vec<Box<dyn Feature>> {
     vec![Box::new(SummaryCommentFeature::new())]
 }
 
+lazy_static! {
+    static ref MUTEX: Mutex<()> = Mutex::new(());
+}
+
 async fn emit_event(
     ctx: &Context,
     event: GitHubEvent,
     data: web::Json<serde_json::Value>,
 ) -> Result<()> {
+    let _guard = MUTEX.lock().unwrap();
```

Comment:
Just to clarify, that this is "bad practice" (https://actix.rs/docs/server#multi-threading), but certainly seems applicable for our use case.

Reply:
Also, might need to un-poisson the mutex if the thread crashes?

Reply:
Also, :sweat_smile: 


```
$ cargo clippy 
warning: this `MutexGuard` is held across an `await` point
  --> src/main.rs:85:9
   |
85 |     let _guard = MUTEX.lock().unwrap();
   |         ^^^^^^
   |
   = note: `#[warn(clippy::await_holding_lock)]` on by default
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> src/main.rs:85:5
   |
85 | /     let _guard = MUTEX.lock().unwrap();
86 | |
87 | |     for feature in features() {
88 | |         if feature.meta().events().contains(&event) {
...  |
93 | |     Ok(())
94 | | }
   | |_^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock

warning: `webhook_features` (bin "webhook_features") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.34s

Reply:
clippy is right. This will fix the bug, but deadlock.

Maybe switch to tokio to fix the poison and deadlock issue?

Reply:
Fixed deadlock and poison in 0a70040a078d61b56792f10a954edfbdc1e273fb

Reply:
Thanks for fixing this.